Packs/AgariPhishingDefense/Integrations/AgariPhishingDefense/AgariPhishingDefense.py

```diff
@@ -0,0 +1,812 @@
+    'PROXY_ERROR': 'Proxy Error - cannot connect to proxy. Either try clearing the \'Use system proxy\' check-box or '
+                   'check the host, authentication details and connection details for the proxy.',
+    'SSL_CERT_ERROR': 'SSL Certificate Verification Failed - try selecting \'Trust any certificate\' checkbox in the '
+                      'integration configuration.',
+    'INTERNAL_SERVER_ERROR': 'The server encountered an internal error for Agari and was unable to complete '
+                             'your request.',
+    'MISSING_SCHEMA_ERROR': 'Invalid API URL. No schema supplied: http(s).',
+    'INVALID_SCHEMA_ERROR': 'Invalid API URL. Supplied schema is invalid, supports http(s).',
+    'INVALID_API_URL': 'Invalid API URL.',
+    'CONNECTION_ERROR': 'Connectivity failed. Check your internet connection or the API URL.',
+    'NO_RECORDS_FOUND': 'No {} were found for the given argument(s).',
+    'FETCH_LIMIT_VALIDATION': 'Value of Fetch Limit must be a positive integer between 1 to 200.',
+    'REQUEST_TIMEOUT': 'Request timed out. Check the configured HTTP(S) Request Timeout (in seconds) value.',
+    'INVALID_TIME_VALIDATION': 'The given value for {0} argument is invalid.',
+    'INVALID_POLICY_ACTION_TYPE': 'The given value for Policy Actions is invalid. Expected "deliver", "mark-spam", '
+                                  '"move", "inbox", "delete" or "none". ',
+    'INVALID_LIMIT': 'Argument limit must be a positive integer between 1 to {}.',
+    'INVALID_PAGE_ID': 'Argument page_id must be a positive integer.',
+    'INVALID_REM_FIELDS': 'Cannot pass "id" in rem_fields argument.',
+    'MISSING_REMEDIATE_ARGS': 'Invalid argument value. Requires both "id" and "operation" argument.',
+    'INVALID_EXCLUDE_ALERT_TYPE': 'The given value for Exclude Alerts is invalid. Expected "System Alert" or '
+                                  '"Message Alert". '
+}
+
+HR_MESSAGES: Dict[str, str] = {
+    'REMEDIATE_MSG_SUCCESS': "Message ID - {} remediated successfully with operation '{}'."
+}
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    def __init__(self, base_url: str, verify: bool, proxy: bool, request_timeout: int, payload):
+        """
+        Initialization of HTTP request timeout, payload(i.e client id and client secret)
+        Initialization of lock object and create a token for the first time if not available in context
+
+        :param base_url: Base url of Endpoint
+        :param verify: Whether the request should verify the SSL certificate.
+        :param proxy: Whether to run the integration using the system proxy.
+        :param request_timeout: Time of Request Timeout
+        :param payload: Payload data (i.e client id and client secret)
+        """
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy)
+        self.request_timeout = request_timeout
+        self.payload = payload
+        self.lock = threading.Lock()
```

Comment:
@guykeller 
We have used this to make the token-refresh part of the code thread-safe since we have parallelism implemented. (usually, most implementations leveraging parallelism requires some kind of thread-safe mechanism with "locks" being the most common solution)